src/helpers.rs

```diff
@@ -1017,6 +1017,35 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         Ok((true, string_length))
     }
 
+    /// Read a sequence of wchar_t until the first null terminator.
+    /// Always returns a `Vec<u32>` no matter the size of `wchar_t`.
+    fn read_wchar_t_str(
```

Comment:
Would it make sense to share the code with `read_wide_str`, and have a single function that can read a sequence of integers until hitting a 0?

Reply:
Definitely. I will try do that when the tests succeeds.